File: app_kms/lib/view/model/assetList.dart
class AssetList {
  String? id;
  String? deviceName;
  String? buyDate;
  String? description;
  String? warrantyStart;
  String? warrantyEnd;
  String? supplier;
  String? poName;
  String? createdBy;
  String? ram;
  String? capacity;
  String? processor;
  String? serialNumber;
  String? department;

  AssetList(
      {this.id,
      this.deviceName,
      this.buyDate,
      this.description,
      this.warrantyStart,
      this.warrantyEnd,
      this.supplier,
      this.poName,
      this.createdBy,
      this.ram,
      this.capacity,
      this.processor,
      this.serialNumber,
      this.department});

  AssetList.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    deviceName = json['device_name'];
    buyDate = json['buy_date'];
    description = json['description'];
    warrantyStart = json['warranty_start'];
    warrantyEnd = json['warranty_end'];
    supplier = json['supplier'];
    poName = json['po_name'];
    createdBy = json['created_by'];
    ram = json['ram'];
    capacity = json['capacity'];
    processor = json['processor'];
    serialNumber = json['serial_number'];
    department = json['department'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['device_name'] = this.deviceName;
    data['buy_date'] = this.buyDate;
    data['description'] = this.description;
    data['warranty_start'] = this.warrantyStart;
    data['warranty_end'] = this.warrantyEnd;
    data['supplier'] = this.supplier;
    data['po_name'] = this.poName;
    data['created_by'] = this.createdBy;
    data['ram'] = this.ram;
    data['capacity'] = this.capacity;
    data['processor'] = this.processor;
    data['serial_number'] = this.serialNumber;
    data['department'] = this.department;
    return data;
    
  }
}

File: app_kms/php/asset_list/asset_list.php

<?php

include_once("dbconnect.php");

$sqlloadproduct = "SELECT id,device_name,buy_date,description,warranty_start,warranty_end,supplier,po_name,created_by,ram,capacity,processor,serial_number,department
FROM asset_list ORDER BY id";

$result = $conn->query($sqlloadproduct);
if ($result->num_rows > 0) {
    $response["assetList"] = array();
    while ($row = $result->fetch_assoc()) {
        $prlist = array();
        $prlist['id'] = $row['id'];
        $prlist['device_name'] = $row['device_name'];
        $prlist['buy_date'] = $row['buy_date'];
        $prlist['description'] = $row['description'];
        $prlist['warranty_start'] = $row['warranty_start'];
        $prlist['warranty_end'] = $row['warranty_end'];
        $prlist['supplier'] = $row['supplier'];
        $prlist['po_name'] = $row['po_name'];
        $prlist['created_by'] = $row['created_by'];
        $prlist['ram'] = $row['ram'];
        $prlist['capacity'] = $row['capacity'];
        $prlist['processor'] = $row['processor'];
        $prlist['serial_number'] = $row['serial_number'];
        $prlist['department'] = $row['department'];
        array_push($response["assetList"],$prlist);
    }
    $response = array('codeResponse' => 200,'status' => 'success', 'data' => $response["assetList"],'count'=>$result->num_rows);
    sendJsonResponse($response);
}else{
$response = array('codeResponse' => -100,'status' => 'failed', 'data' => null,'count'=>0);
sendJsonResponse($response);
}



    

function sendJsonResponse($sentArray)
{
header('Content-Type: application/json');
echo json_encode($sentArray);
}
?>